### PR TITLE
Add exec to helm provider

### PR DESCRIPTION
**Pull request by awcheney**
Adding the 'exec' option to the helm provider that is now in the kubernetes provider. This allows us to create kubeconfigs on the fly.

### CODE REVIEW

**Comment by awcheney** [comment]
@meyskens - could you please take a look at this one?

**Comment by meringu** [comment]
This is copied directly from the Kubernetes provider: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/db2d274250d5e5bf4ea32b8b4a879d2c5e1daee3/kubernetes/provider.go#L107-L134. Can someone please review?

**Comment by meringu** [comment]
@meyskens - could you please take a look at this one?

**Comment by AlexGodard** [comment]
@meyskens - could you please take a look at this one?

**Comment by aaronmell** [comment]
FYI, this won't be supported in Helm V3. The V3 api only supports loading the config from a filepath.

**Comment by meringu** [comment]
Hi @aaronmell, What API are you referencing? Looks like an action.Configuration can be initialized. It's also what the tests do: https://github.com/helm/helm/blob/release-v3.0.0-beta.4/pkg/action/action.go#L66

**Comment by pib** [comment]
Any chance this will be merged soon? I'd love to be able to stop using a custom-built patched version of the provider with EKS.

File: helm/provider.go
Comment (alexsomesan) [inline helm/provider.go]: Would you please fill something descriptive in this Description field?
It makes little sense to include it with an empty string.

**Comment by alexsomesan** [comment]
Change looks good. Minor nit about empty description.
I will test it while waiting for that fix.

**Comment by meringu** [comment]
Thanks @alexsomesan, I've added a description.

**Comment by meyskens** [comment]
Restarted the CI as the acceptance tests seem to be flakey again. The code looks good.
It would also be nice to have the documentation updated in this PR IMO.

**Comment by meringu** [comment]
Thanks @meyskens, I've rebased and added copied the docs from the Kubernetes provider over.

**Review by meyskens — APPROVED** [review]
Looks good to me! Waiting on @alexsomesan for a re-review before merging

**Comment by johnsaltarelli** [comment]
@alexsomesan @meyskens is there an ETA on when this will be merged and released?

**Comment by rgardam** [comment]
Whats the story here guys?

**Comment by jrhouston** [comment]
👋 There are conflicts that need to be resolved here since helm 3 support was added.

**Comment by jrhouston** [comment]
It looks like this feature was actually added when we introduced Helm 3, the docs just weren't updated.

**Comment by nick4fake** [comment]
@jrhouston well, the new version doesn't support Helm2 that is still used in many places, so technically this is not resolved

**Comment by meringu** [comment]
I'm happy for this to be resolved. Given how fast helm 2 support is being dropped and the project no longer supports new features for v2.

Looks like the docs have been updated here https://github.com/terraform-providers/terraform-provider-helm/pull/469 the website just needs to be released.

**Pull request closed without merging**
